modules/ROOT/pages/tools/neo4j-admin/neo4j-admin-import.adoc

```diff
@@ -236,7 +236,7 @@ For an example, see <<import-tool-multiple-input-files-regex-example>>.
 |
 
 |--normalize-types[=true\|false]
-|Whether or not to normalize property types to Cypher types, e.g. `int` becomes `long` and `float` becomes `double`.
+|When true, non-array property values will be converted to their equivalent Cypher types. For example, all integer values will be converted to 64-bit long integers. This takes up more space on disk, but avoids Cypher converting the type during queries.
```

Comment:
> This takes up more space on disk, but avoids Cypher converting the type during queries.

I think @tinwelint mentioned this might not be the case, no?

Did you managed to check this by testing different runs of import with normalization on/off and the difference on the disk-size used?

Reply:
It's handled slightly differently in different formats, e.g. record format(s) for example any integral type, except `long` if it's very large, will fit in one "block" i.e. 8B. Block format tho has dynamic compression based on integral value size, not its type (even works for negative values). So saying that storing primitive integrals as 64-bit long integers instead of their respective type takes up more space is mostly incorrect.

This is what I meant yesterday on the call that storing single primitive integrals doesn't increase store size, whereas doing this to arrays will.

Reply:
@tinwelint is this what you're saying?

1) On Record Format: a byte, short, integer, and long, will all be stored in an 8B block regardless.
2) On Block Format: the number 1 will be stored in a little space as possible regardless of its typing as byte, short, integer or long.

The above I think help explain the thinking behind normalize types. So we should drop the part about saving space, and simply note it simplifies things for Cypher.

Reply:
Yes you're right I think that's good to not involve the space aspect in the description since at least for integral types the only outlier is a negative long in record format(s). Perhaps there's some difference for float/double, but to venture into it would probably just complicate matters.